xonsh/parsers/base.py

```diff
@@ -2098,13 +2098,13 @@ def _attach_nocomma_tok_rules(self):
     # The following grammar rules are no-ops because we don't need to glue the
     # source code back together piece-by-piece. Instead, we simply look for
     # top-level commas and record their positions. With these positions and
-    # the bounding parantheses !() positions we can use the source_slice()
+    # the bounding parenthesis !() positions we can use the source_slice()
```

Comment:
I think this wants to be plural possessive (i.e., "the bounding parentheses' respective positions").